events/models.py

```diff
@@ -64,7 +64,9 @@ class DataSource(models.Model):
     id = models.CharField(max_length=100, primary_key=True)
     name = models.CharField(verbose_name=_('Name'), max_length=255)
     api_key = models.CharField(max_length=128, blank=True, default='')
-    owner = models.ForeignKey('django_orghierarchy.Organization', related_name='owned_systems', null=True, blank=True)
+    owner = models.ForeignKey(
+        'django_orghierarchy.Organization', on_delete=models.CASCADE,
```

Comment:
Here, I would be on the safe side and use SET_NULL so datasources don't get removed.

Reply:
I fixed this in the rebased branch.